cloud-yarn/src/main/java/com/sequenceiq/cloudbreak/cloud/yarn/loadbalancer/service/launch/LoadBalancerLaunchService.java

```diff
@@ -0,0 +1,259 @@
+        ApplicationDetailRequest applicationDetailRequest = new ApplicationDetailRequest();
+        applicationDetailRequest.setName(applicationName);
+        ResponseContext responseContext;
+
+        try {
+            responseContext = yarnClient.getApplicationDetail(applicationDetailRequest);
+        } catch (MalformedURLException ex) {
+            LOGGER.error("Failed to get information for the Yarn loadbalancer! Application name: " + applicationName + " Error: " + ex.getMessage());
+            throw new CloudConnectorException("Failed to get information for the Yarn loadbalancer.", ex);
+        }
+
+        if (responseContext.getStatusCode() == YarnResourceConstants.HTTP_SUCCESS) {
+            ApplicationDetailResponse applicationDetailResponse = (ApplicationDetailResponse) responseContext.getResponseObject();
+            return applicationDetailResponse.getContainers();
+        } else {
+            LOGGER.error("Failed to get yarn container! Application name: " + applicationName);
+            ApplicationErrorResponse errorResponse = responseContext.getResponseError();
+            throw new CloudConnectorException(String.format("Failed to get yarn container details: HTTP Return: %d Error: %s",
+                    responseContext.getStatusCode(), errorResponse == null ? "unknown" : errorResponse.getDiagnostics()));
+        }
+    }
+
+    /**
+     * Gets the names of the instance groups that are Gateway types, the names can then be compared against
+     * Yarn container names to check whether a container is a gateway container or not.
+     */
+    private Set<String> getGatewayGroupNames(CloudStack cloudStack) {
+        return cloudStack.getGroups().stream().filter(group -> InstanceGroupType.isGateway(group.getType()))
+                .map(Group::getName).collect(Collectors.toSet());
+    }
+
+    /**
+     * Creates a Yarn component for each of the loadbalancers that are to be created, using a set of pre-defined
+     * constants for the various component parameters.
+     *
+     * Each container is pointed at a specific docker image which contains the loadbalancing service and logic.
+     */
+    private List<YarnComponent> createLoadBalancerComponents(CloudStack cloudStack, List<String> gatewayIPs, String applicationName) {
+        LOGGER.debug("Creating the loadbalancer components for application " + applicationName + " with gatewayIPs: " + gatewayIPs.toString() + ".");
+        Artifact artifact = createLoadBalancerArtifact();
+        Resource resource = createLoadBalancerResource();
+        String launchCommand = createLoadBalancerLaunchCommand(cloudStack);
+        Configuration configuration = createLoadBalancerConfiguration(gatewayIPs);
+
+        List<YarnComponent> loadBalancerComponents = Lists.newArrayList();
+        for (CloudLoadBalancer loadBalancer : cloudStack.getLoadBalancers()) {
+            String componentName = applicationName + loadBalancer.getType().name();
```

Comment:
Is this the other place you're setting the LB name? Referring to my comment in `YarnMetadataCollector.collectLoadBalancer`. Yeah, please make some shared method both of these can call to generate the LB name. I realize it's just a simple string concatenation, but we don't want these to end up mismatched.